lpc55_sign/src/bin/lpc55_sign.rs

```diff
@@ -545,3 +517,69 @@ fn check_plain_image(_image: &[u8]) -> Result<()> {
     check!(OK, "Nothing to check for plain image");
     Ok(())
 }
+
+fn read_certs(paths: &[PathBuf]) -> Result<Vec<Vec<u8>>> {
+    Ok(paths
+        .iter()
+        .map(std::fs::read)
+        .collect::<Result<Vec<Vec<u8>>, _>>()?)
+}
+
+fn write_to_file(path: &PathBuf, bytes: &[u8]) -> Result<()> {
+    Ok(OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .open(&path)?
+        .write_all(&bytes)?)
+}
+
+fn write_signed_image(
```

Comment:
I'd suggest splitting this into multiple smaller functions which operate in memory, something like

```rust
fn sign_image(
    image: &[u8],
    address: u32,
    signing_certs: &[&[u8]],
    root_certs: &[&[u8]],
    private_key: &[u8],
) -> Vec<u8>
```

```rust
fn generate_cmpa(
    dice: &DiceArgs,
    root_certs: &[&[u8]]
) -> CMPAPage
```

```rust
fn generate_cfpa(
    root_certs: &[&[u8]]
) -> CFPAPage
```

The three operations are philosophically distinct, and we may want to do 1/2/3 of them separately.

This will duplicate some work in generating the RKTH, so you'd probably want to factor that into
```rust
fn calculate_rkth(root_certs: &[&[u8]]) -> [u8; 32]
```

Reply:
Excellent suggestion, thank you. I was trying too hard to merge the `ChainedImage` and `SignedImage` arms; lmk if you think a8b5677 goes too far the other way, but it's pretty close to what you proposed.

Reply:
This also makes `cmpa` an optional long argument like `cfpa`; I guess I understand why in theory we might treat them differently, but it seemed confusing to continue to do so if we're breaking argument compatibility anyway.